docs/guides/auth.md

```diff
@@ -0,0 +1,80 @@
+---
+title: Authentication
+description: "Learn how you can provide granular access to types and fields in GraphQL schema with the help of authentication providers"
+---
+
+## What is Authentication?
+
+Authentication in GraphQL refers to the process of verifying the identity of a user before allowing access to specific fields in GraphQL schema. Essentially, it's about controlling who can see or interact with certain pieces of data exposed by GraphQL API. This process is crucial for following reasons:
+
+- **Protect Sensitive Information** Some fields may contain sensitive information that should not be accessible to unauthenticated users.
+- **Prevent Unauthorized Access** Ensuring that authenticated users can query certain fields helps prevent unauthorized access and potential data breaches.
+- **Customized User Experiences** Authentication allows for personalized queries based on the user's identity.
+
+The authentication could be implemented by different credential validation mechanisms, like:
+
+- **Basic Authentication** is a straightforward authentication scheme that sends base64-encoded usernames and passwords in the HTTP Authorization header with each request. It's simple to implement but requires HTTPS to ensure security due to its lack of encryption.
+- **JWT Authentication** uses digitally signed tokens to authenticate and transmit user information in a compact JSON format, allowing stateless and secure communication between clients and servers. It offers greater flexibility and security, supporting expiration times and custom data embedding within the token itself.
```

Comment:
Should have two separate guides about JWT and Basic Auth.

Reply:
Have a section for authentication and then inside it have three files — Introduction, JWT and then Basic Auth

Reply:
Also need to expand on how auth work for mutations. Provide examples.